sklearn/feature_extraction/tests/test_image.py

```diff
@@ -39,6 +39,14 @@ def test_grid_to_graph():
     A = grid_to_graph(n_x=size, n_y=size, mask=mask, return_as=np.ndarray)
     assert connected_components(A)[0] == 2
 
+    # check ordering
+    mask = np.zeros((2, 3), dtype=np.bool)
+    mask[0, 0] = 1
+    mask[:, 2] = 1
+    graph = grid_to_graph(2, 3, 1, mask=mask.ravel()).todense()
+    assert graph[1, 0] == 0
```

Comment:
Since it will be a 3x3 adjacency matrix, we could use `assert_array_equal` to be sure that all entries are correct.